be/src/http/action/stream_load.cpp

```diff
@@ -299,23 +309,44 @@ void StreamLoadAction::on_chunk_data(HttpRequest* req) {
     auto evbuf = evhttp_request_get_input_buffer(ev_req);
 
     int64_t start_read_data_time = MonotonicNanos();
-    while (evbuffer_get_length(evbuf) > 0) {
-        ByteBufferPtr bb = ByteBuffer::allocate(4096);
+
+    size_t len = 0;
+    while ((len = evbuffer_get_length(evbuf)) > 0) {
+        if (ctx->buffer == nullptr) {
+            // Initialize buffer.
+            ctx->buffer = ByteBuffer::allocate(ctx->kDefaultBufferSize);
+
+        } else if (ctx->buffer->remaining() < len) {
```

Comment:
Also, the file using the gzip/bzip2/lz4/deflate/zstd, the file should be complete.

Reply:
Since the json parser need complete json as input, the allocation of memory whose size is equal to file size is necessary. Maybe some other measure should be taken to avoid allocating too much memory simultaneously.